src/Adapter/Presenter/Product/ProductLazyArray.php

```diff
@@ -462,6 +500,90 @@ public function getEcotax()
         return null;
     }
 
+    /**
+     * @arrayAccess
+     *
+     * @return string|null
+     */
+    public function getManufacturerName()
+    {
+        if (!isset($this->product['manufacturer_name'])) {
+            // Assign empty value
+            $this->product['manufacturer_name'] = null;
+
+            // If we have manufacturer ID, we will try to load it's name and assign it
+            if (!empty($this->product['id_manufacturer'])) {
+                $manufacturerName = Manufacturer::getNameById((int) $this->product['id_manufacturer']);
+                if (!empty($manufacturerName)) {
+                    $this->product['manufacturer_name'] = $manufacturerName;
+                }
+            }
+        }
+
+        return $this->product['manufacturer_name'];
+    }
+
+    /**
+     * @arrayAccess
+     *
+     * @return string|null
+     */
+    public function getCategory()
+    {
+        if (!isset($this->product['category'])) {
+            $categoryLinkRewrite = Category::getLinkRewrite(
+                (int) $this->product['id_category_default'], (int) $this->language->id
+            );
+            $this->product['category'] = !empty($categoryLinkRewrite) ? $categoryLinkRewrite : null;
+        }
+
+        return $this->product['category'];
+    }
```

Comment:
hummm maybe rename the function to `getCategoryLinkRewrite` ?

Reply:
That would be better, but it would be a BC break, because you could no longer access it by $product['category'], is it worth it?